### ISSUE TITLE
Calculating Hash of CommentNode causes nullPointerException

### ISSUE THREAD

**Issue by codesForLiving** [thatJavaNerd/JRAW#145]
It specifically crashes when calculating hash of root `CommentNode`

In file Thing.java

```
public int hashCode() {
        return getId().hashCode();
    }
```

The `getId()` returns null for root `CommentNode`

**Comment by fbis251** [comment]
Hey,

I originally wrote this code, do you have any recommendations on what value to return by default?

Since we're doing a hashCode for a String object, it looks like the default reutrn value for a [String's hash code is 0](
http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/String.java#l117) so we could just return that by default

**Comment by codesForLiving** [comment]
Yes, returning 0 would be the good default behavior. As, in CommentNode.java, inside 'hashCode()'

`        result = 31 * result + (comment != null ? comment.hashCode() : 0);`

here also if `comment==null`, we return 0.

Yes, I was inserting them in Set. Currently I have to use Map<String,CommentNode> as workAround.
And Key here is `CommentNode..getComment().getId()`

**Comment by thatJavaNerd** [comment]
Closed with commit [`165f57e`]
